File: vipclaw-admin/src/main/java/com/vipamp/vipclaw/admin/entity/SysJobLog.java
package com.vipamp.vipclaw.admin.entity;

import com.baomidou.mybatisplus.annotation.*;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 定时任务日志实体类
 *
 * @author vipamp
 * @since 2026-03-16
 */
@Data
@EqualsAndHashCode(callSuper = false)
@TableName("sys_job_log")
@Schema(description = "定时任务日志实体类")
public class SysJobLog implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * 日志ID
     */
    @TableId(value = "id", type = IdType.AUTO)
    @Schema(description = "日志ID")
    private Long id;

    /**
     * 任务ID
     */
    @Schema(description = "任务ID")
    private Long jobId;

    /**
     * 任务名称
     */
    @Schema(description = "任务名称")
    private String jobName;

    /**
     * 任务组名
     */
    @Schema(description = "任务组名")
    private String jobGroup;

    /**
     * 调用目标
     */
    @Schema(description = "调用目标")
    private String invokeTarget;

    /**
     * 执行信息
     */
    @Schema(description = "执行信息")
    private String jobMessage;

    /**
     * 执行状态（0-失败，1-成功）
     */
    @Schema(description = "执行状态（0-失败，1-成功）")
    private Integer status;

    /**
     * 异常信息
     */
    @Schema(description = "异常信息")
    private String exceptionInfo;

    /**
     * 开始时间
     */
    @Schema(description = "开始时间")
    private LocalDateTime startTime;

    /**
     * 结束时间
     */
    @Schema(description = "结束时间")
    private LocalDateTime endTime;

    /**
     * 创建时间
     */
    @Schema(description = "创建时间")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/admin/entity/SysJobLog.kt
package com.vipamp.vipclaw.admin.entity

import io.swagger.v3.oas.annotations.media.Schema
import java.io.Serializable
import java.time.LocalDateTime

@Schema(description = "Scheduled job log entity")
class SysJobLog : Serializable {
    companion object {
        private const val serialVersionUID = 1L
    }

    @Schema(description = "Log ID")
    var id: Long = 0

    @Schema(description = "Job ID")
    var jobId: Long = 0

    @Schema(description = "Job name")
    var jobName: String = ""

    @Schema(description = "Job group")
    var jobGroup: String = ""

    @Schema(description = "Invocation target")
    var invokeTarget: String = ""

    @Schema(description = "Execution message")
    var jobMessage: String = ""

    @Schema(description = "Execution status (0-failure, 1-success)")
    var status: Int = 1

    @Schema(description = "Exception information")
    var exceptionInfo = ""

    @Schema(description = "Start time")
    var startTime: LocalDateTime = LocalDateTime.now()

    @Schema(description = "End time")
    var endTime: LocalDateTime = LocalDateTime.now()

    @Schema(description = "Creator")
    var creator: String = ""

    @Schema(description = "Creation time")
    var createTime: LocalDateTime = LocalDateTime.now()
}
